config/v1/types_build.go

```diff
@@ -32,54 +32,83 @@ type BuildSpec struct {
 }
 
 type BuildDefaults struct {
+	// RegistryProxy is the proxy settings for image pull/push
+	// +optional
+	RegistryProxy Proxy `json:"registryProxy,omitempty"`
 	// GitHTTPProxy is the location of the HTTPProxy for Git source
 	// +optional
 	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
-
 	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
 	// +optional
 	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
-
-	// GitNoProxy is the list of domains for which the proxy should not be used
+	// GitNoProxy is the list of domains for which the proxy should not be used for Git source
 	// +optional
 	GitNoProxy string `json:"gitNoProxy,omitempty"`
-
 	// Env is a set of default environment variables that will be applied to the
 	// build if the specified variables do not exist on the build
 	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
-
 	// ImageLabels is a list of docker labels that are applied to the resulting image.
 	// User can override a default label by providing a label with the same name in their
 	// Build/BuildConfig.
 	// +optional
 	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
-
 	// Resources defines resource requirements to execute the build.
 	// +optional
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// RegistriesConfig controls the registries allowed for image pull and push.
+	// +optional
+	RegistriesConfig RegistriesConfig `json:"registriesConfig,omitempty"`
 }
 
 type ImageLabel struct {
 	// Name defines the name of the label. It must have non-zero length.
 	Name string `json:"name"`
-
 	// Value defines the literal value of the label.
 	// +optional
 	Value string `json:"value,omitempty"`
 }
 
+type Proxy struct {
+	// HTTPProxy is the URL of this proxy for HTTP traffic
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+	// HTTPSProxy is the URL of this proxy for HTTPS traffic
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+	// NoProxy is the list of the domains for which this proxy should not be used
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
+
+type RegistriesList []string
+
+var (
+	DefaultSearchRegistries RegistriesList = []string{"registry.redhat.io", "quay.io", "registry.access.redhat.com", "docker.io"}
```

Comment:
We disabled these defaults in 3.x - why would we need to shave them here?

Reply:
yes, just say no to registry search lists.  embrace the fully qualified image pull spec.  we painted ourselves into a corner by doing this in the past.

https://bugzilla.redhat.com/show_bug.cgi?id=1561989

Reply:
Nm, this is search.

Search is pretty fraught.  I'm not sure I want to support that in the future.  We agreed in 3.10 / 3.11 that search is counter to our goals with Kubernetes and so endorsing and future tolerating global search is counter to our goals.

Reply:
If this has been (and would continue to be) a pain point, how big of a pain would we create by removing this as a confg option?

I think we're in agreement that the sane default is `nil`, but I'm hesitant to remove search as a config option.